pkg/controller/jobs/jobset/jobset_controller.go

```diff
@@ -121,7 +121,7 @@ func (j *JobSet) PodSets() ([]kueue.PodSet, error) {
 	podSets := make([]kueue.PodSet, len(j.Spec.ReplicatedJobs))
 	for index, replicatedJob := range j.Spec.ReplicatedJobs {
 		podSets[index] = kueue.PodSet{
-			Name:     replicatedJob.Name,
+			Name:     kueue.PodSetReference(replicatedJob.Name),
```

Comment:
```suggestion
			Name:     kueue.PodSetReference(strings.ToLower(replicatedJob.Name)),
```
Maybe like this? WDYT?